src/Sylius/Bundle/CoreBundle/Form/Type/Product/ChannelPricingType.php

```diff
@@ -13,25 +13,49 @@
 
 use Sylius\Bundle\MoneyBundle\Form\Type\MoneyType;
 use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
 use Sylius\Component\Core\Model\ChannelInterface;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\Form\FormEvent;
 use Symfony\Component\Form\FormEvents;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
  */
 final class ChannelPricingType extends AbstractResourceType
 {
     /**
+     * @var ChannelRepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param ChannelRepositoryInterface $channelRepository
+     */
+    public function __construct(
+        $dataClass,
+        $validationGroups = [],
+        ChannelRepositoryInterface $channelRepository
+    ) {
+        parent::__construct($dataClass, $validationGroups);
+
+        $this->channelRepository = $channelRepository;
+    }
+
+    /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $channelRepository = $this->channelRepository;
+
         $builder
-            ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) {
+            ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($channelRepository, $options) {
                 /** @var ChannelInterface $channel */
-                $channel = $event->getData()->getChannel();
+                $channel = isset($options['channel']) ? $options['channel'] : $channelRepository->findOneByCode($event->getData()->getChannelCode());
```

Comment:
Unfortunatelly yes, and [this](https://github.com/Sylius/Sylius/pull/7590/files/29fc336bb1427499979d3dffc914f6563b7dce4e#diff-62ad2217f324e2930dee4427f25052e5R80) is the case.